connectors/src/api/webhooks/webhook_slack.ts

```diff
@@ -371,7 +377,86 @@ const _webhookSlackAPIHandler = async (
         }
         break;
       }
+      case "channel_created": {
+        const teamId = req.body.team_id;
+        const slackChannelId = req.body.event?.channel;
+        const slackConfiguration = await SlackConfigurationModel.findOne({
+          where: {
+            slackTeamId: teamId,
+          },
+        });
+        if (!slackConfiguration || !slackChannelId) {
+          return apiError(req, res, {
+            api_error: {
+              type: "slack_configuration_not_found",
+              message: `Slack configuration not found for teamId ${teamId}`,
+            },
+            status_code: 404,
+          });
+        }
+        const connectorId = slackConfiguration.connectorId;
 
+        const connector = await ConnectorResource.fetchById(connectorId);
+        if (!connector) {
+          return apiError(req, res, {
+            api_error: {
+              type: "connector_not_found",
+              message: `Connector ${req.params.connector_id} not found`,
+            },
+            status_code: 404,
+          });
+        }
+        const slackClient = await getSlackClient(connectorId);
+        const remoteChannel = await slackClient.conversations.info({
+          channel: slackChannelId,
+        });
+        const remoteChannelName = remoteChannel.channel?.name;
+
+        if (!remoteChannel.ok || !remoteChannelName) {
+          logger.error({
+            connectorId,
+            channelId: slackChannelId,
+            error: remoteChannel.error,
+          });
+          return new Err(
+            new Error("Could not get the Slack channel information.")
+          );
+        }
+
+        const whiteListedChannelPatterns =
+          slackConfiguration.whiteListedChannelPatterns;
+        const isWhiteListed = isChannelNameWhitelisted(
+          remoteChannelName,
+          whiteListedChannelPatterns
+        );
+        if (isWhiteListed) {
+          try {
+            await SlackChannel.create({
+              connectorId: connectorId,
+              slackChannelId: slackChannelId,
+              slackChannelName: remoteChannelName,
+              permission: "read_write",
+            });
+          } catch (error) {
+            console.error({
+              connectorId,
+              channelId: slackChannelId,
+              error: remoteChannel.error,
+            });
+            return new Err(new Error("Could not create the Slack channel."));
+          }
+          const joinChannelRes = await joinChannel(
+            connectorId,
+            slackChannelId
+          );
+          if (joinChannelRes.isErr()) {
+            throw new Error(
```

Comment:
We generally prefer not to throw errors. Instead, we use a `Result` return type or `apiError` on API endpoints.

Reply:
yess. 1. from https://dust.tt/w/0ec9852c2f/assistant/b202ec479f

Reply:
Some discussion here: https://app.slack.com/client/T050RH73H9P/C050SM8NSPK

Reply:
Please read it and make sure you understand the invariants :pray:

Reply:
I really need to write a doc on this

Reply:
Added a small blob in coding rules on https://www.notion.so/dust-tt/Engineering-e0f834b5be5a43569baaf76e9c41adf2